siliconcompiler/core.py

```diff
@@ -438,42 +409,50 @@ def find_function(self, modulename, functype, funcname):
         * flows/modulename.py
         * pdks/modulname.py
 
+        If the modtype is None, the module paths are search in the
+        order: 'targets'->'flows'->'tools'->'pdks'->'libs'):
+
+
         Supported functions include:
 
-        * pdk (make_docs, setup_pdk)
-        * flow (make_docs, setup_flow)
-        * tool (make_docs, setup_tool, check_version, runtime_options,
+        * targets (make_docs, setup)
+        * pdks (make_docs, setup)
+        * flows (make_docs, setup)
+        * tools (make_docs, setup, check_version, runtime_options,
           pre_process, post_process)
+        * libs (make_docs, setup)
 
         Args:
             modulename (str): Name of module to import.
-            functype (str): Type of function to import (tool,flow, pdk).
             funcname (str): Name of the function to find within the module.
+            modtype (str): Type of module (tools,flows, pdks,libs,targets).
 
         Examples:
-            >>> setup_pdk = chip.find_function('freepdk45','pdk','setup_pdk')
+            >>> setup_pdk = chip.find_function('freepdk45', 'setup', 'pdk')
             >>> setup_pdk()
             Imports the freepdk45 module and runs the setup_pdk function
 
         '''
 
-        # module search path depends on functype
-        if functype == 'tool':
-            fullpath = self._find_sc_file(f"tools/{modulename}/{modulename}.py", missing_ok=True)
-        elif functype == 'flow':
-            fullpath = self._find_sc_file(f"flows/{modulename}.py", missing_ok=True)
-        elif functype == 'pdk':
-            fullpath = self._find_sc_file(f"pdks/{modulename}.py", missing_ok=True)
-        elif functype == 'project':
-            fullpath = self._find_sc_file(f"projects/{modulename}.py", missing_ok=True)
+        # module search path depends on modtype
+        if moduletype is None:
+            for item in ('targets', 'flows', 'pdks', 'libs'):
```

Comment:
Should this also include 'tools' between 'flows' and 'pdks', or should there be a special case for 'tools' like with the statements below?